src/components/seo.tsx

```diff
@@ -2,16 +2,69 @@ import Head from 'next/head';
 
 type Props = {
   pageTitle: string;
+  pageMeta?: {
+    nftName: string;
+    nftAuthor: string;
+    nftDescription: string;
+    nftImage: string;
+  };
 };
 
-const SEO = ({ pageTitle }: Props) => (
+const SEO = ({ pageTitle, pageMeta }: Props) => (
   <Head>
     <title> {pageTitle} | LEDA - Jupiter Based NFT Marketplace</title>
+    <link rel="icon" href="/favicon.ico" />
     <meta httpEquiv="x-ua-compatible" content="ie=edge" />
-    <meta name="description" content="Generated by create next app" />
     <meta name="robots" content="noindex, follow" />
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
-    <link rel="icon" href="/favicon.png" />
+
+    {pageMeta ? (
```

Comment:
Please create separate components for these two cases. When the component starts to grow is a good candidate to be refactored into smaller components.